### PR TITLE
[metricsadvisor] convert to test proxy

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves https://github.com/Azure/azure-sdk-for-python/issues/21126

- Updates tests to use test proxy
- Removes dependency on unittest
- Combines API key and AAD tests with pytest.parametrize
- Adds better randomization so we can run live tests in parallel, rather than sequentially
- Fixes error handling so we see exceptions in CI
- Updated secrets for CI and gets our live test pipeline green again :)

### CODE REVIEW

File: sdk/metricsadvisor/azure-ai-metricsadvisor/tests/async_tests/base_testcase_async.py [hunk 1]
```diff
@@ -48,179 +36,99 @@
     WebNotificationHook,
 )
 
+# for pytest.parametrize
+subscription_key = os.getenv("METRICS_ADVISOR_SUBSCRIPTION_KEY", "metrics_advisor_subscription_key")
```
Comment (xiangyan99) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/tests/async_tests/base_testcase_async.py:40]: In https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/metricsadvisor/tests.yml, we set 
AZURE_METRICS_ADVISOR_SUBSCRIPTION_KEY
but here we use 
METRICS_ADVISOR_SUBSCRIPTION_KEY

Is it by design?

Comment (kristapratico) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/tests/async_tests/base_testcase_async.py:40]: The tests.yml is updated in this PR to drop the AZURE_ prefix: https://github.com/Azure/azure-sdk-for-python/blob/1d3cb9c3934752376ee2064bb9988e16543c1dc5/sdk/metricsadvisor/tests.yml

Main motivations were to shorten the names and more closely match the naming convention we use for env vars with the PowershellPreparer work (yes I know we don't use PowershellPreparer in this library yet, but if we ever do it the renames will be done).

Comment (xiangyan99) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/tests/async_tests/base_testcase_async.py:40]: Does this (shortening the env name) also apply to other libraries?

Comment (kristapratico) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/tests/async_tests/base_testcase_async.py:40]: If using the PowershellPreparer, the env vars must fit this convention for them to be properly sanitized so it does apply in that case. I don't have the context of why we decided to do it like that. If you want to keep AZURE_ for this library, I can revert those renames.

Comment (xiangyan99) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/tests/async_tests/base_testcase_async.py:40]: I don't have a strong preference to have AZURE_ prefix or not. But I would like to keep consistency across our libraries then we can add it into our guideline.

One of my concerns is some env names may be too generic if we don't have the AZURE_ prefix. (Maybe that's the reason we added it?)

Comment (kristapratico) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/tests/async_tests/base_testcase_async.py:40]: I can't link to it exactly, but in our tests doc under 3) we do say to create the env vars with this naming convention: https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/tests.md#create-live-test-resources

File: sdk/metricsadvisor/azure-ai-metricsadvisor/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+# Ignore async tests for Python < 3.5
+collect_ignore_glob = []
+if sys.version_info < (3, 5):
+    collect_ignore_glob.append("*_async.py")
+
+
+@pytest.fixture(scope="session", autouse=True)
+def add_sanitizers(test_proxy):
+    service_endpoint = os.getenv("METRICS_ADVISOR_ENDPOINT", "https://fakeendpoint.cognitiveservices.azure.com")
```
Comment (xiangyan99) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/tests/conftest.py]: service_endpoint = service_endpoint.rstrip("/")

File: sdk/metricsadvisor/azure-ai-metricsadvisor/tests/test_metrics_advisor_client_live.py [hunk 3]
```diff
@@ -100,68 +127,95 @@ def test_list_metrics_series_data(self):
...
+    @recorded_by_proxy
+    def test_list_metric_series_definitions(self, client):
+        results = list(client.list_metric_series_definitions(
             metric_id=self.metric_id,
             active_since=datetime.datetime(2021, 1, 1),
         ))
         assert len(results) > 0
 
-    def test_add_anomaly_feedback(self):
+    @pytest.mark.parametrize("credential", API_KEY, ids=ids)  # only using API key for now since service issue with AAD
```
Comment (xiangyan99) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/tests/test_metrics_advisor_client_live.py]: Could you tell me why here we only run upon api_key?

Comment (kristapratico) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/tests/test_metrics_advisor_client_live.py]: Was seeing BadRequest returned from the service like what karishma reported [here](https://teams.microsoft.com/l/message/19:11b7faed215e4d22bfe8b60d4a716474@thread.skype/1636749642057?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1636749642057&teamName=Azure%20SDK&channelName=Service%20-%20Metrics%20Advisor&createdTime=1636749642057)

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**